src/fastpm.c

```diff
@@ -1080,10 +1077,12 @@ usmesh_ready_handler(FastPMUSMesh * mesh, FastPMLCEvent * lcevent, struct usmesh
         fastpm_info("Appending usmesh catalog to %s\n", filebase);
         fastpm_store_write(lcevent->p, filebase, "a", prr->cli->Nwriters, fastpm->comm);
     }
-    write_aemit_hist(filebase, "1/.", data->cdm_hist, fastpm->comm);
+    char * dataset_attrs = fastpm_strdup_printf("%s/.", data->tail->name);
+    write_aemit_hist(filebase, dataset_attrs, data->par_hist, fastpm->comm);
+    free(dataset_attrs);
 
     /* halos */
-    if(CONF(prr->lua, write_fof)) {
+    if(CONF(prr->lua, write_fof) && *data->tail->name == *fastpm_species_get_name(FASTPM_SPECIES_CDM)) {
```

Comment:
These equality checks feel dangers if name is a string then this only checks the equality of the pointer not the string itself. I recall tail is of type FastPMStore -- what about saving the species enum as a member of FastPMStore and then you can check data->tail->species == FASTPM_SPECIES_CDM?